### PR TITLE
updates `metrics.yaml` with correct `income` type for new schema; changes name of income question in legacy data to `exactIncome`

### PR DESCRIPTION
**Pull request by hamilton**
Some of the work toward #621. 

This PR changes the metric type for our opt-in demographic survey question around income from a set of range bands to a single quantity.

Specifically this {R:

- updates `metrics.yaml` to change the metric type for `income` to `quantity`
- updates the "survey schema" we've been using to use the key `incomeExact` instead of `income` since it needs to be named differently than the already-exising `income`. There will be an accompanying PR in the current schema to match this change but we want to get this in asap since we will be launching two new study partners soon and need to publish this web extension on AMO first.

@Dexterp37 tagging you for a review of the changes on the Glean end – could you take a look at the metrics change and let me know if I forgot anything?

@rhelmer tagging you for the rest of the review.

Checklist for reviewer:

- [ ] The description should reference a bug or github issue, if relevant.
- [ ] There must be a [`CHANGELOG.md`](./CHANGELOG.md) entry for any non-test change.
- [ ] Any change to the NPM commands must be carefully reviewed to make sure it won't break the Add-ons pipeline.
- [ ] Any version increase must follow the [release process](./docs/RELEASE_PROCESS.md).

### CODE REVIEW

File: metrics.yaml [hunk 1]
```diff
@@ -181,22 +181,15 @@ user:
...
-      - band_100000_149999
-      - band_ge_150000
     description: |
       The user household's combined annual income during the
       past 12 months.
     bugs:
-      - https://github.com/mozilla-rally/rally-core-addon/issues/545
+      - https://github.com/mozilla-rally/rally-core-addon/issues/621
     data_reviews:
       - https://github.com/mozilla-rally/rally-core-addon/pull/139#issuecomment-736024232
```
Comment (hamilton) [inline metrics.yaml:194]: I am assuming this line will have to be updated with the data review (probably a github comment on this PR) approving this.

Comment (Dexterp37) [reply metrics.yaml:195]: Yes

File: metrics.yaml [hunk 2]
```diff
@@ -181,22 +181,16 @@ user:
     expires: never
 
   income:
```
Comment (Dexterp37) [inline metrics.yaml:183]: Since you're changing the type, you should rename this: maybe `raw_income` or `exact_income`?

**Review by Dexterp37 — CHANGES_REQUESTED** [review]
The call-site of the Glean API should be changed as well: the old APIs do not exist anymore (if you do a clean rebuild it will throw)

**Review by rhelmer — COMMENTED** [review]
I think @Dexterp37's review is both necessary and sufficient :)

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,7 +5,9 @@
 # v1.3.1 (2021-05-20)
 
 [Full changelog](https://github.com/mozilla-rally/core-addon/compare/v1.3.0...v1.3.1)
-* [#610](https://github.com/mozilla-rally/rally-core-addon/pull/610): Downgrade protocol, not working on taskcluster
+
+* [#610](https://github.com/mozilla-rally/rally-core-addon/pull/610): Downgrade protocol, not working on taskcluster.
+* [#624](https://github.com/mozilla-rally/rally-core-addon/pull/624): Update the `income` field to use the key `incomeExact`; update `metrics.yaml` to change `income` to `quantity`.
```
Comment (Dexterp37) [inline CHANGELOG.md:10]: This should be above, in the `Unrealeased changes` section.

File: core-addon/DataCollection.js [hunk 4]
```diff
@@ -339,6 +338,10 @@ export default class DataCollection {
       processed["zipCode"] = data["zipcode"];
     }
 
+    if ("exactIncome" in data) {
+      processed['exactIncome'] = data["exactIncome"];
```
Comment (Dexterp37) [inline core-addon/DataCollection.js:342]: Don't forget to update the schema for legacy telemetry :(

**Comment by Dexterp37** [comment]
> 
> 
> @Dexterp37 Thank you for the initial review – updated the PR and re-upped the review request. How do I move forward with a data review on this?

Sure! You can copy the template you can find https://wiki.mozilla.org/Data_Collection#Requesting_Data_Collection as a comment in this PR (or in the related issue), the flag somebody on the reviewer's list on that page.

File: CHANGELOG.md (hunk 3 above)
Comment (hamilton) [reply CHANGELOG.md:10]: whoops! Thanks for the catch.

File: core-addon/DataCollection.js (hunk 4 above)
Comment (hamilton) [reply core-addon/DataCollection.js:342]: indeed – will be filing a PR for this as well.

**Comment by hamilton** [comment]
# Request for Data Collection Renewal

** This form is for the renewal of an existing, reviewed data collection.**

** All questions are mandatory.
You must receive Data Review from a
[Data Steward](https://wiki.mozilla.org/Data_Collection)
on a filled-out Request before shipping your renewed data collection.**

1) Provide a link to the initial Data Collection Review Request for this collection.
2) When will this collection now expire?
3) Why was the initial period of collection insufficient?

File: metrics.yaml (hunk 1 above)
Comment (travis79) [reply metrics.yaml:195]: I recommend just adding it to the list and not changing the bug numbers

Comment (travis79) [reply metrics.yaml:195]: Finding the earlier bugs for reference and context is why this is a list

**Comment by hamilton** [comment]
# Request for data collection review form

NOTE: previous review [here](https://github.com/mozilla-rally/rally-core-addon/pull/139#issuecomment-730959560).

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

1) What questions will you answer with this data?

Our academic partners use the demographic survey to answer research questions. It is an optional part of the Rally experience. The income question will enable them to answer research questions relevant to income.

2) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements? Some example responses:

Mozilla needs to answer these questions because they are a part of our partner & customer requirements. Partners will not likely work with us or will have to create workarounds to get the same data; we want to offer this as part of the opt-in Rally experience.

3) What alternative methods did you consider to answer these questions? Why were they not sufficient?

There are no alternatives to directly asking a Rally panel member these demographic questions.

4) Can current instrumentation answer these questions?

No. Rally has already been collecting this demographic data, but we are changing our income question (technically expanding it to have a different type) so that we can better meet researchers' needs.

5) List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Data_Collection) found on the Mozilla wiki.

**Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.**

The list of collected metrics is available in the `DATA_DOCUMENTATION.md` file. Here's a [convenient link](https://github.com/mozilla-rally/rally-core-addon/blob/559babc5e2e3a458578fe6a5f02d0c6a841a7adc/docs/DATA_COLLECTION.md#filling-the-demographic-survey-demographic-survey) to the relevant section.

The above is probably either Category 3 or Category 4. I'm not sure which one, though.

6) Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.

Documentation for these metrics is part of this PR (`DATA_DOCUMENTATION.md` and will permanently live in the `docs/` directory of this repository).

7) How long will this data be collected?  Choose one of the following:

Permanently unless the user opts to delete the data. @knowtheory and @rjweiss are responsible for this data.

8) What populations will you measure?

The population of users joining the Rally platform.

9) If this data collection is default on, what is the opt-out mechanism for users?

It is not default on.

10) Please provide a general description of how you will analyze this data.

The data will be used to statistically describe the panel of users joining the Rally platform.

11) Where do you intend to share the results of your analysis?

Aggregate reports, anonymous statistics created from the collected data may be published with research papers from our partners, on the Rally website or internally.

12) Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection? If so:

No. This is all Mozilla-run or created technology for collection and analysis.

**Comment by Dexterp37** [comment]
> 
> 
> Please update your version of NPM and drop the changes to the lockfile: I think you're using an old NPM version and that's rewriting the full lockfile to v1.

@hamilton don't forget about this :)

**Comment by hamilton** [comment]
@Dexterp37 done, thanks for the reminder.

**Review by Dexterp37 — APPROVED** [review]

**Comment by travis79** [comment]
> # Request for data collection review form
> 
> NOTE: previous review [here](https://github.com/mozilla-rally/rally-core-addon/pull/139#issuecomment-730959560).
> 
> **All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**
> 
>     1. What questions will you answer with this data?
> 
> 
> Our academic partners use the demographic survey to answer research questions. It is an optional part of the Rally experience. The income question will enable them to answer research questions relevant to income.
> 
>     1. Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements? Some example responses:
> 
> 
> Mozilla needs to answer these questions because they are a part of our partner & customer requirements. Partners will not likely work with us or will have to create workarounds to get the same data; we want to offer this as part of the opt-in Rally experience.
> 
>     1. What alternative methods did you consider to answer these questions? Why were they not sufficient?
> 
> 
> There are no alternatives to directly asking a Rally panel member these demographic questions.
> 
>     1. Can current instrumentation answer these questions?
> 
> 
> No. Rally has already been collecting this demographic data, but we are changing our income question (technically expanding it to have a different type) so that we can better meet researchers' needs.
> 
>     1. List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Data_Collection) found on the Mozilla wiki.
> 
> 
> **Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.**
> 
> The list of collected metrics is available in the `DATA_DOCUMENTATION.md` file. Here's a [convenient link](https://github.com/mozilla-rally/rally-core-addon/blob/559babc5e2e3a458578fe6a5f02d0c6a841a7adc/docs/DATA_COLLECTION.md#filling-the-demographic-survey-demographic-survey) to the relevant section.
> 
> The above is probably either Category 3 or Category 4. I'm not sure which one, though.
> 
>     1. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
> 
> 
> Documentation for these metrics is part of this PR (`DATA_DOCUMENTATION.md` and will permanently live in the `docs/` directory of this repository).
> 
>     1. How long will this data be collected?  Choose one of the following:
> 
> 
> Permanently unless the user opts to delete the data. @knowtheory and @rjweiss are responsible for this data.
> 
>     1. What populations will you measure?
> 
> 
> The population of users joining the Rally platform.
> 
>     1. If this data collection is default on, what is the opt-out mechanism for users?
> 
> 
> It is not default on.
> 
>     1. Please provide a general description of how you will analyze this data.
> 
> 
> The data will be used to statistically describe the panel of users joining the Rally platform.
> 
>     1. Where do you intend to share the results of your analysis?
> 
> 
> Aggregate reports, anonymous statistics created from the collected data may be published with research papers from our partners, on the Rally website or internally.
> 
>     1. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection? If so:
> 
> 
> No. This is all Mozilla-run or created technology for collection and analysis.

# Data Review

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 

Yes, in the `DATA_DOCUMENTATION.md` file in this repository.

2) Is there a control mechanism that allows the user to turn the data collection on and off? 

Yes, by opting out of the study, opting out of Rally, or by disabling telemetry.

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

Yes, Ted Han and Rebecca Weiss

4) Using the **[category system of data types](https://wiki.mozilla.org/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Category 4: Highly sensitive or clearly identifiable personal data.  

5) Is the data collection request for default-on or default-off?

Default off, the user needs to opt into both Rally and the specific study for the collection to occur.

6) Does the instrumentation include the addition of **any *new* identifiers**?

No

7) Is the data collection covered by the existing Firefox privacy notice? 

Yes, and the privacy policies covering Rally and the individual study.

8) Does the data collection use a third-party collection tool? 

No

## Result

data-review+

**Pull request merged**